src/internet_identity/src/storage.rs

```diff
@@ -324,6 +336,69 @@ impl<T: candid::CandidType + serde::de::DeserializeOwned, M: Memory> Storage<T,
         }
         Ok(record_number)
     }
+
+    /// Returns the address of the first byte not yet allocated to a user.
+    /// This address exists even if the max user number has been reached, because there is a memory
+    /// reserve at the end of stable memory.
+    fn unused_memory_start(&self) -> u64 {
+        let record_number = self.header.num_users as u64;
+        RESERVED_HEADER_BYTES + record_number * self.header.entry_size as u64
+    }
```

Comment:
why don't we simply always return the value after the 8 gigs allocated to users? In production we have many more users than on test canisters; by always assuming the canister may be full we reduce the risk of issues in production that we can't trigger with test canisters with fewer users